src/librustc/mir/interpret/value.rs

```diff
@@ -445,6 +451,13 @@ impl<'tcx, Tag> Scalar<Tag> {
         Ok(b as i8)
     }
 
+    pub fn to_i16(self) -> InterpResult<'static, i16> {
+        let sz = Size::from_bits(16);
+        let b = self.to_bits(sz)?;
+        let b = sign_extend(b, sz) as i128;
+        Ok(b as i16)
+    }
+
```

Comment:
Why not have `to_signed_with_bit_width` (?) and then rewrite this as:
```rust
self.to_signed_with_bit_width(16).map(|v| v as i16)
```
(Same with the other types and unsigned variants.)

Reply:
done :D